SqlInstall.ps1

```diff
@@ -0,0 +1,556 @@
+﻿function Install-SqlServer ([string]$AppVolume, [string]$DataVolume, [string]$LogVolume, [string]$TempVolume, [string]$BackupVolume, [int]$Ctp)
+{
+    <#
+    .SYNOPSIS
+
+    This function will help you to quickly install a SQL Server instance. 
+
+    .DESCRIPTION
+
+    This function will help you to quickly install a SQL Server instance. By scanning your system it will apply a number of best practices to your installation
+    The max DOP and max server memory will be set, the sa account will be disabled while the account used to install SQL Server will be added as sysadmin.
+    The number of TempDB files will be set to the number of cores with a maximum of eight.
+    The perform volume maintenance right will be granted to the SQL Server account. If you run this script in an environment where you are not allowed to do this,
+    please revert that operation by removing the right from the local security policy (secpol.msc).
+
+    The autogrowth of file will be set to 64MB, the Cost Threshold for Parallelism will be set to 40. 
+
+    Note that the dowloaded installation file must be unzipped, an ISO has to be mounted. This will not be executed from this script.
+    
+    .PARAMETER Name
+
+    $AppVolume will hold the volume letter of the application disc. If left empty, it will default to C, unless there is a drive named like App
+
+    $DataVolume will hold the volume letter of the Data disc. If left empty, it will default to C, unless there is a drive named like Data
+
+    $LogVolume will hold the volume letter of the Log disc. If left empty, it will default to C, unless there is a drive named like Log
+
+    $TempVolume will hold the volume letter of the Temp disc. If left empty, it will default to C, unless there is a drive named like Temp
+
+    $BackupVolume will hold the volume letter of the Backup disc. If left empty, it will default to C, unless there is a drive named like Backup
+
+    $Ctp will hold the value of your Cost Threshold for Parallelism. If left empty, it will default to 40 (the mean between OLTP(30) and Datawarehousing(50))
+
+    .Inputs
+    None
+
+    .Outputs
+    None
+
+    .Example
+    C:\PS> Install-SqlServer
+
+    This will run the installation with the default settings
+
+    C:\PS> Install-SqlServer -AppVolume "G"
+
+    This will run the installation with default setting apart from the application volume, this will be redirected to the G drive.
+
+
+    #>
+
+    $startScript = @'
+
+;SQL Server 2014 Configuration File
+[OPTIONS]
+; Specifies a Setup work flow, like INSTALL, UNINSTALL, or UPGRADE. This is a required parameter. 
+ACTION="Install"
+; Detailed help for command line argument ROLE has not been defined yet. 
+ROLE="AllFeatures_WithDefaults"
```

Comment:
AllFeatures_WithDefaults is a heavy default. I'd place $Role as a parameter in the script and set it to whatever database engine only is.  So someone can do `Install-DbaInstance -Role AllFeaturesWithDefaults` if they'd like.

Reply:
Missed that one... From an Express perspective it won't be too bad, but from an Enterprise edition some weird stuff could be happening.